src/resonate/scheduler.py

```diff
@@ -109,19 +136,28 @@ def __init__(
         self._scheduler = scheduler
         self._threads = set[Thread]()
 
-    def enqueue(self, sqe: _SQE[Any]) -> None:
+    def enqueue(self, sqe: _SQE) -> None:
         self._sq.put_nowait(sqe)
         self._adjust_thread_count()
 
     def _run(self) -> None:
         while True:
             sqe = self._sq.dequeue()
-            fn_result = sqe.fn.run()
-            assert isinstance(fn_result, (Ok, Err)), f"{fn_result} must be a result."
-            self._scheduler.enqueue_cqe(
-                sqe=sqe,
-                value=fn_result,
-            )
+            if isinstance(sqe, _FnSQE):
+                fn_result = sqe.fn.run()
+                assert isinstance(
+                    fn_result, (Ok, Err)
+                ), f"{fn_result} must be a result."
+                self._scheduler.enqueue_cqe(_FnCQE(sqe, fn_result))
+            elif isinstance(sqe, _BatchSQE):
+                result: Result[list[Any], Exception]
+                try:
+                    result = Ok(sqe.handler(list(sqe.commands)))
+                except Exception as e:  # noqa: BLE001
+                    result = Err(e)
```

Comment:
Should we fail the whole batch because a single command failed or should we return a list of "fails" and "oks" so the user can know which commands to retry?

Reply:
Then I don't think we can have a list of "oks" either, this batch operations will probably return a single result as well, and then will all the promises for all the Commands have the same result? For example: "Upload all these images to S3" the result will be "all  images uploaded" so there is no multiple results for it return.

Reply:
There API. That allow you to summit an array of thinks to process and return an array of results (one per submitted object)

Using a list support that. Handlers can optionally return a list or a single result. If it's a single result all promises will be resolved with the same result. If it's a list we resolve using the promise on the same position.

But that's a requirement

Reply:
U were right. It makes sense to have an element level success or failure. I update the code to account for it